From 14597d7cbae01039021b1f8265c2e94183ac2000 Mon Sep 17 00:00:00 2001
From: Jason Chen <b02280@freescale.com>
Date: Wed, 9 Nov 2011 15:12:31 +0800
Subject: [PATCH 0589/2463] ENGR00161757 imx: add i2c depends of some drivers

add i2c depends for mxc edid/dvi/hdmi driver.
add i2c depends for codec soc sgtl5000 and cs42888.

Signed-off-by: Jason Chen <b02280@freescale.com>
---
 drivers/video/Kconfig      |    2 +-
 drivers/video/mxc/Kconfig  |    7 ++++++-
 drivers/video/mxc/Makefile |    2 +-
 sound/soc/imx/Kconfig      |    4 ++--
 4 files changed, 10 insertions(+), 5 deletions(-)

diff --git a/drivers/video/Kconfig b/drivers/video/Kconfig
index 3811e48..7ca56f9 100644
--- a/drivers/video/Kconfig
+++ b/drivers/video/Kconfig
@@ -2394,7 +2394,7 @@ source "drivers/video/mxc/Kconfig"
 endif
 
 config FB_MXC_HDMI
-	depends on FB_MXC_SYNC_PANEL
+	depends on FB_MXC_SYNC_PANEL && I2C
 	tristate "MXC HDMI driver support"
 	select MFD_MXC_HDMI
 	help
diff --git a/drivers/video/mxc/Kconfig b/drivers/video/mxc/Kconfig
index 1c548d0..fe9de73 100644
--- a/drivers/video/mxc/Kconfig
+++ b/drivers/video/mxc/Kconfig
@@ -14,6 +14,11 @@ config FB_MXC
 	  If you plan to use the LCD display with your MXC system, say
 	  Y here.
 
+config FB_MXC_EDID
+	depends on FB_MXC && I2C
+	tristate "MXC EDID support"
+	default y
+
 config FB_MXC_SYNC_PANEL
 	depends on FB_MXC
 	tristate "Synchronous Panel Framebuffer"
@@ -52,7 +57,7 @@ config FB_MXC_SEIKO_WVGA_SYNC_PANEL
 	tristate "SEIKO WVGA Panel"
 
 config FB_MXC_SII902X
-	depends on FB_MXC_SYNC_PANEL
+	depends on FB_MXC_SYNC_PANEL && I2C
 	tristate "Si Image SII9022 DVI/HDMI Interface Chip"
 
 config FB_MXC_CH7026
diff --git a/drivers/video/mxc/Makefile b/drivers/video/mxc/Makefile
index debae23..084837c 100644
--- a/drivers/video/mxc/Makefile
+++ b/drivers/video/mxc/Makefile
@@ -3,7 +3,7 @@ obj-$(CONFIG_FB_MXC_SII902X)                    += mxcfb_sii902x.o
 obj-$(CONFIG_FB_MXC_LDB)			+= ldb.o
 obj-$(CONFIG_FB_MXC_MIPI_DSI)			+= mipi_dsi.o
 obj-$(CONFIG_FB_MXC_TRULY_WVGA_SYNC_PANEL) 	+= mxcfb_hx8369_wvga.o
-obj-$(CONFIG_FB_MODE_HELPERS)			+= mxc_edid.o mxc_dvi.o
+obj-$(CONFIG_FB_MXC_EDID)			+= mxc_edid.o mxc_dvi.o
 ifeq ($(CONFIG_ARCH_MX21)$(CONFIG_ARCH_MX27)$(CONFIG_ARCH_MX25),y)
 	obj-$(CONFIG_FB_MXC_TVOUT)              += fs453.o
 	obj-$(CONFIG_FB_MXC_SYNC_PANEL)         += mx2fb.o mxcfb_modedb.o
diff --git a/sound/soc/imx/Kconfig b/sound/soc/imx/Kconfig
index 3ea833d..9bb8b93d 100644
--- a/sound/soc/imx/Kconfig
+++ b/sound/soc/imx/Kconfig
@@ -52,7 +52,7 @@ config SND_SOC_PHYCORE_AC97
 
 config SND_SOC_IMX_SGTL5000
 	tristate "SoC Audio support for i.MX boards with sgtl5000"
-	depends on MACH_MX35_3DS || MACH_MX51_BABBAGE || MACH_MX6Q_SABRELITE
+	depends on I2C && (MACH_MX35_3DS || MACH_MX51_BABBAGE || MACH_MX6Q_SABRELITE)
 	select SND_SOC_SGTL5000
 	select SND_MXC_SOC_MX2
 	help
@@ -61,7 +61,7 @@ config SND_SOC_IMX_SGTL5000
 
 config SND_SOC_IMX_CS42888
 	tristate "SoC Audio support for i.MX boards with cs42888"
-	depends on MACH_MX6Q_ARM2 || MACH_MX53_ARD
+	depends on I2C && (MACH_MX6Q_ARM2 || MACH_MX53_ARD)
 	select SND_SOC_CS42888
 	select SND_MXC_SOC_MX2
 	help
-- 
1.7.7.4

